prog/alloc.go

```diff
@@ -51,13 +51,17 @@ func (ma *memAlloc) noteAlloc(addr0, size0 uint64) {
 	}
 }
 
-func (ma *memAlloc) alloc(r *randGen, size0 uint64) uint64 {
+func (ma *memAlloc) alloc(r *randGen, size0 uint64, alignment uint64) uint64 {
 	if size0 == 0 {
 		size0 = 1
 	}
+	if alignment == 0 {
+		alignment = 1
+	}
 	size := (size0 + memAllocGranule - 1) / memAllocGranule
+	alignment = (alignment + memAllocGranule - 1) / memAllocGranule
```

Comment:
Could you explain this?
If I understand correctly, alignments <= 64 become 1, bigger alignments become roughly alignment/64.

Reply:
We've discussed this offline. I've missed that size is also divided by memAllocGranule.